httpcore/_backends/sync.py

```diff
@@ -78,8 +78,8 @@ def close(self) -> None:
                 self.sock.close()
 
     def is_connection_dropped(self) -> bool:
-        rready, _wready, _xready = select.select([self.sock], [], [], 0)
-        return bool(rready)
+        rready = wait_for_read([self.sock], timeout=0)
```

Comment:
May you please add the same mechanism to the `curio` backend?

https://github.com/encode/httpcore/blob/685a9b6c968f0662b45f637ffacaf4795454fb9d/httpcore/_backends/curio.py#L135-L138